### PR TITLE
Add NET6 support

### LINKED ISSUE #6: Bind Nuke with ios .net6 project
To be able to use Nuke with MAUI, we need to copy the Xamarin.Nuke project, and use ios-net6 target.
Then adapt the github actions to compile and publish the package to nuget.

@Cheesebaron do you think you will have time for that? Thanks pal :)

### PR DESCRIPTION
**Pull request by Cheesebaron**
This PR adds .NET6.0 support. I unfortunately had to create a new NuGet package for .NET6.0 as handrolling the package is not easy and would require a lot of steps.

Switching to xcframework broke the old binding, and I don't really have the patience to support this legacy. So users wanting to use Nuke on xamarin.ios targets are stuck on old versions 🤷‍♂️

As agreed in the comments the package has also now been renamed to ImageCaching.Nuke

Fixes #6

### CODE REVIEW

File: Xamarin.Nuke.nuspec [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <authors>Jean-Marie Alfonsi, Tomasz Cielecki</authors>
     <owners>roubachof,cheesebaron</owners>
     <license type="file">LICENSE</license>
-    <projectUrl>https://github.com/roubachof/Xamarin.Nuke</projectUrl>
+    <projectUrl>https://github.com/roubachof/Xamarin.Nuke.NET6</projectUrl>
```
Comment (roubachof) [inline Xamarin.Nuke.nuspec]: The url is in fact https://github.com/roubachof/NukeProxy

File: Xamarin.Nuke/Xamarin.Nuke.NET6/Xamarin.Nuke.NET6.csproj [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    <IsBindingProject>true</IsBindingProject>
+    <NoBindingEmbedding>true</NoBindingEmbedding>
+    <RootNamespace>Xamarin.Nuke</RootNamespace>
+    <SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-ios'">14.0</SupportedOSPlatformVersion>
+    <SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-maccatalyst'">14.0</SupportedOSPlatformVersion>
+  </PropertyGroup>
+  
+  <PropertyGroup>
+    <Copyright>Sharpnado 2020</Copyright>
+    <PackageProjectUrl>https://github.com/roubachof/Xamarin.Nuke</PackageProjectUrl>
```
Comment (roubachof) [inline Xamarin.Nuke/Xamarin.Nuke.NET6/Xamarin.Nuke.NET6.csproj]: The url is in fact https://github.com/roubachof/NukeProxy

**Review by roubachof — COMMENTED** [review]
Wow !!!! This is so great, thanks for this!!! 
I think we should drop the `Xamarin` prefix since its there is no xamarin sauce is .net 6.
Maybe something like Nuke.Net6 ?

File: Xamarin.Nuke.nuspec (hunk 1 above)
Comment (Cheesebaron) [reply Xamarin.Nuke.nuspec]: Yep, thanks. Fixed :)

File: Xamarin.Nuke/Xamarin.Nuke.NET6/Xamarin.Nuke.NET6.csproj (hunk 2 above)
Comment (Cheesebaron) [reply Xamarin.Nuke/Xamarin.Nuke.NET6/Xamarin.Nuke.NET6.csproj]: Fixed this, thanks

**Comment by Cheesebaron** [comment]
We could change the name. That is super easy to do.

**Comment by roubachof** [comment]
Yes I think we should. Do you have a better idea than Nuke.NET6?

**Comment by Cheesebaron** [comment]
Not really, there is already another project called Nuke, which is a build system. Perhaps that would confuse some people.

Also not sure if they've reserved the Nuke.* namespace, so we might not be able to use it.

**Comment by roubachof** [comment]
True... ImageCaching.Nuke.Net ?

**Comment by roubachof** [comment]
Simply ImageCaching.Nuke ?
🤔

**Comment by Cheesebaron** [comment]
ImageCaching.Nuke could work. Descriptive of what it is.

**Comment by roubachof** [comment]
Ok let's try this

File: src/Sample/Info.plist [hunk 3]
```diff
@@ -2,66 +2,66 @@
...
-        <string>UIInterfaceOrientationPortraitUpsideDown</string>
-        <string>UIInterfaceOrientationLandscapeLeft</string>
-        <string>UIInterfaceOrientationLandscapeRight</string>
-    </array>
-    <key>XSAppIconAssets</key>
-    <string>Assets.xcassets/AppIcon.appiconset</string>
+	<key>CFBundleDisplayName</key>
+	<string>Nuke Sample</string>
+	<key>CFBundleIdentifier</key>
+	<string>awesomeorg.Xamarin-Nuke-Sample</string>
```
Comment (roubachof) [inline src/Sample/Info.plist]: maybe change this to awesomeorg.nuke.sample

Comment (Cheesebaron) [reply src/Sample/Info.plist]: Sure

**Comment by Cheesebaron** [comment]
@roubachof good to go?

**Pull request merged**